File: CompetitiveProgramming/DMOJ/TLE/TLE '15 P3 - Poetry/2021-04-29/Poetry.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Poetry
{
	public static void main (String [] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		int[] limits = new int[n];
		
		for(int i = 0; i < n; i++)
			limits[i] = Integer.parseInt(br.readLine());
		
		String line = br.readLine() + " ";
		int first = 0;
		int last = 0;
		for(int i = 0; first < line.length(); i++)
		{
			int limit = limits[i%n];
			last = first + limit;
			last = line.lastIndexOf(" ", last);
			
			if(last < first)
			{
				System.out.println(new String(line.substring(first, first+limit)));
				first += limit;
			}
			else
			{
				System.out.println(new String(line.substring(first, last)));
				first = last + 1;
			}
		}
	}
}

File: CompetitiveProgramming/DMOJ/TLE/TLE '15 P3 - Poetry/2021-04-29/Poetry.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
	cin.sync_with_stdio(0);
	cin.tie(0);

	int n; cin >> n;
	int limits[n];
	
	for(int i = 0; i < n; i++)
		cin >> limits[i];

	cin.ignore(256, '\n');
	string line;
	getline(cin, line);
	line += " ";

	int first = 0;
	int last = 0;

	for(int i = 0; first < line.length(); i++)
	{
		int limit = limits[i%n];
		last = first + limit;
		last = line.find_last_of(' ', last);
		
		if(last < first)
		{
			cout << (line.substr(first, limit)) << endl;
			first += limit;
		}
		else
		{
			cout << (line.substr(first, last-first)) << endl;
			first = last + 1;
		}
	}
	
	return 0;
}